components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/executers/tasks/RemoteFetchConfigurationImmediateTask.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.remotefetch.core.executers.tasks;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchComponentRegistry;
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchConfiguration;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListener;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerComponent;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployer;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerComponent;
+import org.wso2.carbon.identity.remotefetch.common.exceptions.RemoteFetchCoreException;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManager;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerComponent;
+import org.wso2.carbon.identity.remotefetch.core.internal.RemoteFetchServiceComponentHolder;
+
+/**
+ * This class provide immediate task to be executed for trigger call of given remote fetch configuration.
+ * It retrieves relevant action lister component, deployer component and repository manager component using given
+ * remote fetch configuration object from componant registry.
+ * It builds action listener, deployer and repository manager using relevant builders
+ * It runs corresponding action listener iteration. *
+ */
+public class RemoteFetchConfigurationImmediateTask implements Runnable {
+
+    private static final Log log = LogFactory.getLog(RemoteFetchConfigurationImmediateTask.class);
+
+    private RemoteFetchComponentRegistry componentRegistry = RemoteFetchServiceComponentHolder.getInstance()
```

Comment:
Do not use something like this in classLoad time. It slows down the system startup.
instead the componentRegistry needs to be supplied as a constructor argument or as a setter by the caller.